File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int curr = 0;
        int res = 0;
        for(int i = 0; i< s.length(); i++){
            if(s.charAt(i)=='(')
                curr++;
            else if(s.charAt(i)==')')
                curr--;
            res= Math.max(curr, res);
        }
        return res;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        res = 0
        cur = 0
        
        for c in s:
            if c == '(':
                cur += 1
            elif c == ')':
                cur -= 1
            res = max(res,cur)
        return res